packages/gatsby-theme-vtex/src/components/Minicart/Drawer/Delete.tsx

```diff
@@ -0,0 +1,41 @@
+import { Button, Spinner } from '@vtex/store-ui'
+import React, { useState } from 'react'
+
+import { useUpdateItems } from './useUpdateItems'
+
+type Props = {
+  index: number
+  variant?: string
+}
+
+const MinicartDelete = (props: Props) => {
+  const [isLoading, setIsLoading] = useState(false)
+  const updateItems = useUpdateItems(props.index)
+  const useDelete = {
+    variant: `${props.variant}.delete`,
+    onClick: () => {
+      setIsLoading(true)
+      updateItems(0, () => setIsLoading(false))
```

Comment:
What happens if updateItems fail ? Will `setIsLoading` be called ? Can't we use a try...catch approach in here ? Instead of `updateItems` to accept a callback, let it return a promise ? We do something similar in buyButton's onClick method